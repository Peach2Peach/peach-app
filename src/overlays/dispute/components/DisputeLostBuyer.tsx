import React, { ReactElement } from 'react'
import { Text } from '../../../components'
import tw from '../../../styles/tailwind'
import i18n from '../../../utils/i18n'

type DisputeLostBuyerProps = {
  tradeId: string
}

export const DisputeLostBuyer = ({ tradeId }: DisputeLostBuyerProps): ReactElement => (
  <>
    <Text>{i18n('dispute.buyer.lost.text.1', tradeId)}</Text>
    <Text style={tw`mt-3`}>{i18n('dispute.buyer.lost.text.2')}</Text>
  </>
)
