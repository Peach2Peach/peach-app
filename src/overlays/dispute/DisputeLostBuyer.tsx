import React, { ReactElement } from 'react'
import { Text } from '../../components'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'

type DisputeLostBuyerProps = {
  tradeId: string
}

export const DisputeLostBuyer = ({ tradeId }: DisputeLostBuyerProps): ReactElement => (
  <>
    <Text style={tw`mb-3 body-m text-black-1`}>{i18n('dispute.buyer.lost.text.1', tradeId)}</Text>
    <Text style={tw`mb-3 body-m text-black-1`}>{i18n('dispute.buyer.lost.text.2')}</Text>
  </>
)
