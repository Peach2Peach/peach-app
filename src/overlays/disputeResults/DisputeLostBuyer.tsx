import React, { ReactElement, useContext } from 'react'
import { View } from 'react-native'
import { Headline, Text, PrimaryButton } from '../../components'
import { OverlayContext } from '../../contexts/overlay'
import tw from '../../styles/tailwind'
import { saveContract } from '../../utils/contract'
import i18n from '../../utils/i18n'

type DisputeLostBuyerProps = {
  contract: Contract
  navigate: () => void
}

export const DisputeLostBuyer = ({ contract, navigate }: DisputeLostBuyerProps): ReactElement => {
  const [, updateOverlay] = useContext(OverlayContext)

  const closeOverlay = () => {
    saveContract({
      ...contract,
      disputeResultAcknowledged: true,
      cancelConfirmationDismissed: true,
    })
    navigate()
    updateOverlay({ visible: false })
  }

  return (
    <View style={tw`px-6`}>
      <Headline style={tw`text-3xl leading-3xl text-white-1`}>{i18n('dispute.lost')}</Headline>
      <View style={tw`flex items-center justify-center`}>
        <View style={tw`flex items-center justify-center`}>
          <Text style={tw`text-center text-white-1`}>{i18n('dispute.buyer.lost.text.1')}</Text>
          {!contract.paymentConfirmed && (
            <Text style={tw`mt-2 text-center text-white-1`}>{i18n('dispute.buyer.lost.text.2')}</Text>
          )}
        </View>
        <PrimaryButton style={tw`mt-5`} onPress={closeOverlay} narrow>
          {i18n('close')}
        </PrimaryButton>
      </View>
    </View>
  )
}
