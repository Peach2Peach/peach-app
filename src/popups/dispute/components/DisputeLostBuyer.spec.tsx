import { createRenderer } from 'react-test-renderer/shallow'
import { DisputeLostBuyer } from './DisputeLostBuyer'

describe('DisputeLostBuyer', () => {
  const shallowRenderer = createRenderer()
  it('should render correctly', () => {
    shallowRenderer.render(<DisputeLostBuyer tradeId="tradeId" />)
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })
})
