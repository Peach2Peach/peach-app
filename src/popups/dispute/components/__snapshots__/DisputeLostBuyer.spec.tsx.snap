// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisputeLostBuyer should render correctly 1`] = `
<React.Fragment>
  <PeachText>
    We've ruled in favour of the seller in the dispute for trade tradeId.
  </PeachText>
  <PeachText
    style={
      {
        "marginTop": 12,
      }
    }
  >
    Please check the message from your mediator in the trade chat to see what's next.
  </PeachText>
</React.Fragment>
`;
